app/src/main/java/org/mozilla/focus/widget/DownloadListAdapter.java

```diff
@@ -36,14 +39,17 @@ public DownloadListAdapter(){
     public void fetchEntity(){
         mDownloadInfo.clear();
         mDownloadInfo = DBUtils.getDbService().getAllDownloadInfo();
+        Collections.reverse(mDownloadInfo);
         this.notifyDataSetChanged();
     }
 
     public void updateItem(DownloadInfo downloadInfo){
 
         for (int i = 0;i<mDownloadInfo.size();i++){
             if (mDownloadInfo.get(i).getDownloadId().equals(downloadInfo.getDownloadId())){
-                mDownloadInfo.set(i,downloadInfo);
+                mDownloadInfo.remove(i);
+                mDownloadInfo.add(downloadInfo);
```

Comment:
You remove the older one and replace with new one, but what would happen when you add the new one to the bottom of the list?